test/inputs/text-input.spec.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+var expect = require('chai').expect;
+var TextInput = require('../../lib/inputs/text-input');
+var expectError = require('./input-test-utils').expectError;
+
+describe('text input', () => {
+    describe('valid cases', () => {
+        const testObj = {
+            id: 'id',
+            type: 'text',
+            value: 'default value',
+            params: {
+                default: 'default value'
+            }
+        };
+
+        function updateTestObj(update) {
+            return Object.assign({}, testObj, update);
+        }
+
+        it('handles normal text input with default and value', () => {
+            let input = new TextInput('id', testObj.params);
+            expect(input.toObj()).to.deep.equal(testObj);
+
+            input.setValue('other value');
+            expect(input.toObj()).to.deep.equal(updateTestObj({
+                value: 'other value'
+            }));
+        });
+
+        it('not setting value or default has value of emptry string', () => {
```

Comment:
I'd also rephrase the description to be: "builtin default value is an empty string".